pkg/scheduler/flavorassigner/tas_flavorassigner.go

```diff
@@ -57,13 +56,11 @@ func (a *Assignment) WorkloadsTopologyRequests(wl *workload.Info, cq *schdcache.
 }
 
 func (psa *PodSetAssignment) HasFailedNode(wl *workload.Info) bool {
-	if !workload.HasNodeToReplace(wl.Obj) {
-		return false
-	}
-	nodesToReplace := wl.Obj.Status.NodesToReplace
-	for _, domain := range psa.TopologyAssignment.Domains {
-		if slices.Contains(nodesToReplace, domain.Values[len(domain.Values)-1]) {
-			return true
+	if workload.HasUnhealthyNodes(wl.Obj) {
+		for _, domain := range psa.TopologyAssignment.Domains {
+			if workload.HasUnhealthyNode(wl.Obj, domain.Values[len(domain.Values)-1]) {
+				return true
+			}
 		}
 	}
 	return false
```

Comment:
```suggestion
	return workload.HasUnhealthyNodes(wl.Obj) && slices.ContainsFunc(psa.TopologyAssignment.Domains, func(domain kueue.TopologyDomainAssignment) bool {
		return workload.HasUnhealthyNode(wl.Obj, domain.Values[len(domain.Values)-1])
	})
```

Let's make this function more Go idiomatic.

Reply:
Done in https://github.com/kubernetes-sigs/kueue/pull/6822.